apis/v1alpha1/webhook.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2021 The Cockroach Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+var (
+	// DefaultGRPCPort is the default port used for GRPC communication
+	DefaultGRPCPort int32 = 26258
+	// DefaultSQLPort is the default port used for SQL connections
+	DefaultSQLPort int32 = 26257
+	// DefaultHTTPPort is the default port for the Web UI
+	DefaultHTTPPort int32 = 8080
+	// DefaultMaxUnavailable is the default max unavailable nodes during a rollout
+	DefaultMaxUnavailable int32 = 1
+)
+
+var (
+	// log is for logging in this package.
+	webhookLog = logf.Log.WithName("webhooks")
+
+	_ webhook.Defaulter = &CrdbCluster{}
+	_ webhook.Validator = &CrdbCluster{}
+)
+
+// SetupWebhookWithManager ensures webhooks are enabled for the CrdbCluster resource.
+func (r *CrdbCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).For(r).Complete()
+}
+
+//+kubebuilder:webhook:path=/mutate-crdb-cockroachlabs-com-cockroachlabs-com-v1alpha1-crdbcluster,mutating=true,failurePolicy=fail,groups=crdb.cockroachlabs.com.cockroachlabs.com,resources=crdbclusters,verbs=create;update,versions=v1alpha1,name=mcrdbcluster.kb.io,sideEffects=None,admissionReviewVersions=v1alpha1
```

Comment:
I think we can use something less verbose for the path (i.e. `/mutate-crdbcluster-v1alpha1`)